policyengine_us/variables/gov/states/ri/tax/income/adjusted_gross_income/social_security/ri_social_security_modification.py

```diff
@@ -0,0 +1,57 @@
+from policyengine_us.model_api import *
+import numpy as np
+
+
+class ri_social_security_modification(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Rhode Island Social Security Modification"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-12/Social%20Security%20Worksheet_w.pdf"
+    defined_for = "ri_social_security_modification_eligibility"
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        is_head = person("is_tax_unit_head", period)
+        is_spouse = person("is_tax_unit_spouse", period)
+        age = person("age", period)
+        birth_year = period.start.year - age
+
+        p = parameters(
+            period
+        ).gov.states.ri.tax.income.adjusted_gross_income.subtractions.social_security
+
+        # Age-based eligibility.
+        age_conditions = birth_year <= p.birth_date_limit
+        head_eligible = tax_unit.sum(is_head & age_conditions) > 0
+        spouse_eligible = tax_unit.sum(is_spouse & age_conditions) > 0
+
+        # Status eligibility.
+
+        total_social_security = person("social_security", period)
+
+        taxable_social_security = person("taxable_social_security", period)
+
+        head_total_ss = tax_unit.sum(total_social_security * is_head)
+
+        spouse_total_ss = tax_unit.sum(total_social_security * is_spouse)
+
+        head_social_security = where(head_eligible, head_total_ss, 0)
+
+        final_ss = where(
+            spouse_eligible,
+            spouse_total_ss,
+            head_social_security,
+        )
+
+        percentage_social_security = np.zeros_like(head_total_ss)
+        mask = head_total_ss != 0
+        percentage_social_security[mask] = final_ss[mask] / head_total_ss[mask]
```

Comment:
dont you divide by the eligible ss amount and not just head

Reply:
check if one or two people are eligible and if only one divid the amount from the eligible person by the total

Reply:
In https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-12/Social%20Security%20Worksheet_w.pdf, line 8, it says "Amount of social security benefits from Federal Form 1040 or 1040-SR, line 6a", and line 10, "Eligible percentage of social security benefits. Divide line 9 by line 8." I think only line 8, which is the head's social security benefits, is the denominator.

Reply:
I think line 6a is all social security benefits, regardless of head or souse - because if just use head then we get 0.5% every time even if both spouses are eligible